src/Features/CSharp/Portable/AddParameter/CSharpAddParameterCodeFixProvider.cs

```diff
@@ -39,5 +43,49 @@ protected override ImmutableArray<string> TooManyArgumentsDiagnosticIds
 
         protected override ImmutableArray<string> CannotConvertDiagnosticIds
             => GenerateConstructorDiagnosticIds.CannotConvertDiagnosticIds;
+
+        protected override async Task<bool> HandleLanguageSpecificNodesAsync(SyntaxNode initialNode, SyntaxNode node, CodeFixContext context)
```

Comment:
looking at the PR, the bool return value seems to serve no purpose.  i would just remove and simplify the overall flow of logic here.

Reply:
The return value is currently used to exit the loop, but I will try to refactor this anyway.

Reply:
Gotcha.  

That said, if you can do the refactorin to make it so that these methods return the relevant info, rather than doing the registering themselves, then that would be enough to break the loop.  Let me know if that works out!